app.js

```diff
@@ -185,7 +186,48 @@ function MonitorApp ({
       });
     };
     
+    // Store in memcache db
+    // TODO: deprecate memcache db. just use mongo for everything.
     mjs.set(key, JSON.stringify(newRoutes), {}, handleErr);
+    
+    // Add new routes to mongo db log. Used for generating timeseries.
+    db.collection('routeLog').insertOne({
+      'timestamp': now,
+      // can omit timestamp from each route object since they're all the same
+      'routes': newRoutes.map((r) => _.omit(r, 'timestamp'))
+    });
+
+  }));
+
+  app.get('/api/v0/numNodesTimeseries', asyncMiddleware(async function(req, res) {
+    let numNodes = [];
+    let numGateways = [];
+    let timestamps = [];
+    
+    let now = new Date();
+    let yesterday = new Date(now - 1000 * 60 * 60 * 24);
+    let toDate = now;
+    let fromDate = yesterday;
+    if (req.query.from)
+      fromDate = new Date(req.query.from);
```

Comment:
My style (take or leave it, just fyi):

I try to always use `const` and generally limit mutable variables.
Therefore when you see `let x =...` you know to expect `x` to change later.

Assuming equal readability, limit number of lines and intermediary variables so more of the program fits in one screen. So I'd use tertiary operator here like `const fromDate = req.query.from ? new Date(req.query.from) : yesterday`

Reply:
Going to leave it, but it's always interesting to hear / think about other possible styles. Maybe I'll start using const one of these days...